executor/set_test.go

```diff
@@ -502,6 +502,19 @@ func (s *testSuite2) TestValidateSetVar(c *C) {
 	_, err = tk.Exec("set @@global.max_connections='hello'")
 	c.Assert(terror.ErrorEqual(err, variable.ErrWrongTypeForVar), IsTrue)
 
+	tk.MustExec("set @@global.thread_pool_size=65")
+	tk.MustQuery("show warnings").Check(testutil.RowsWithSep("|", "Warning|1292|Truncated incorrect thread_pool_size value: '65'"))
+	result = tk.MustQuery("select @@global.thread_pool_size;")
+	result.Check(testkit.Rows("64"))
```

Comment:
According to https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_thread_pool_size,
> If a value outside this range is specified, the thread pool plugin does not load and the server writes a message to the error log. 

Is it right to correct `thread_pool_size` to 64 here?

Reply:
Hi, @SunRunAway 

Let me see if I understood you correctly. Your concern is the expected value of line 505 should be 64, right?

----

When I reading the source code, I think these tests are covering this kind of scenarios, when people setting an invalid value. The maximum or minimum value will be set.

Take the previous test case as example, the max value of global.max_connection is 100000, when people try to set 100001, it returns 10000.

![image](https://user-images.githubusercontent.com/1224077/65495860-fcb55800-dee9-11e9-9711-6aa916cff7b1.png)



https://github.com/pingcap/tidb/blob/07339a53b0db026cf2313217fb41ced9f73f8f19/executor/set_test.go#L490

I hope I have answered you question.

Thanks.

Reply:
Hi @SunRunAway 

Here is the source code of validation:

https://github.com/pingcap/tidb/blob/e173c7f5c1041b3c7e67507889d50a7bdbcdfc01/sessionctx/variable/varsutil.go#L233

when you set a value that is greater than the maximum value, this function will assign the maximum value to this system variable.

In my case, the maximum value of thread_pool_size is 64, when people assign 65 to it, it will be abandoned, and assign 64 to this system variable.

I hope that answered your question.

Reply:
Thank you for your patient and specific explanation. I get it.